### DISCUSSION TITLE
当选择msvc时 clangd无法正常使用

### DISCUSSION THREAD

**Discussion by RI-ss** [xmake-io/xmake#3592] [Q&A]
![image](https://user-images.githubusercontent.com/71321125/229461498-d722c536-e8bb-45c8-912b-d1eea2525e55.png)
![image](https://user-images.githubusercontent.com/71321125/229461796-3d05c038-cd6c-46f3-99a6-a78be7fb786a.png)
![image](https://user-images.githubusercontent.com/71321125/229461882-5ec9b068-7394-483f-b320-eca344d0e557.png)
![image](https://user-images.githubusercontent.com/71321125/229462160-4ed5aaaa-4cef-4039-b93f-709fd35d0eeb.png)

此时toolchain为clang clangd可以使用

toolchain改为msvc
![image](https://user-images.githubusercontent.com/71321125/229462797-2151f858-ca25-44b0-b8e1-04cd3f18d3e2.png)
![image](https://user-images.githubusercontent.com/71321125/229462918-cd9004cc-5003-4dc0-8661-996aad9cf196.png)
报错

如何在toolchain为msvc下正常使用clangd

**Comment by waruqi** [comment]
自己检查下生成的 compile_commands.json 参数和include路径，如果参数和路径没问题，那就是 clangd 的问题，你得到 clangd 那边反馈，如果路径不对，自己调整下编译配置。
